src/tests/end-to-end/tests/details-view/tabstop.test.ts

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { launchBrowser } from '../../common/browser-factory';
+import { fastPassSelectors } from '../../common/element-identifiers/common-selectors';
+import { popupPageElementIdentifiers } from '../../common/element-identifiers/popup-page-element-identifiers';
+import { Page } from '../../common/page-controllers/page';
+import { TargetPage } from '../../common/page-controllers/target-page';
+import { Browser } from './../../common/browser';
+import { DEFAULT_PAGE_ELEMENT_WAIT_TIMEOUT_MS } from './../../common/timeouts';
+
+describe('Tabstop tests', () => {
+    describe('tabstop from fastpass', () => {
+        let browser: Browser;
+        let targetPage: TargetPage;
+        let fastPassPage: Page;
+
+        beforeAll(async () => {
+            browser = await launchBrowser({ suppressFirstTimeDialog: true });
+            targetPage = await browser.newTargetPage();
+            fastPassPage = await gotoFastPass(browser, targetPage.tabId);
+        });
+
+        afterAll(async () => {
+            if (browser) {
+                await browser.close();
+            }
+        });
+
+        test('if visualHelper is turned on after starting tabstop and pressing tabs on target page', async () => {
+            const toggleAriaLabel = 'Tab stops';
+
+            await goToTabStopTest();
+            await enableToggleByAriaLabel(toggleAriaLabel);
+
+            await targetPage.bringToFront();
+
+            // press tabs 3 times. todo: DRY this code
+            await targetPage.keyPress('Tab');
+            await targetPage.keyPress('Tab');
+            await targetPage.keyPress('Tab');
+
+            expect(await targetPage.getShadowRootHtmlSnapshot()).toMatchSnapshot();
+        });
+
+        async function gotoFastPass(browserLocal: Browser, targetTabId: number): Promise<Page> {
+            const popupPage = await browserLocal.newPopupPage(targetPage);
+            let fastPass: Page;
+            await Promise.all([
+                browserLocal.waitForDetailsViewPage(targetPage).then(page => (fastPass = page)),
+                popupPage.clickSelector(popupPageElementIdentifiers.fastPass),
+            ]);
+
+            return fastPass;
+        }
+
+        async function goToTabStopTest(): Promise<void> {
+            await fastPassPage.clickSelector(fastPassSelectors.tabstopNavButtonSelector);
+            // await fastPassPage.disableAnimations();
+            await verifyTabStopPageLoaded();
+        }
+
+        async function enableToggleByAriaLabel(ariaLabel: string): Promise<void> {
+            const toggleSelector = `button[aria-label="${ariaLabel}"]`;
+            const enabledToggleSelector = `${toggleSelector}[aria-checked=true]`;
+            const disabledToggleSelector = `${toggleSelector}[aria-checked=false]`;
+
+            await fastPassPage.waitForDuration(6000);
```

Comment:
It doesn't seem to be clear by reading this function why we start by waiting for 6000 ms (I presume). Shouldn't we interact with the page and then wait for "something" to happen?

Also, `waitForDuration` seems brittle. `6000` seems like an arbitrary value which may or may not work base on the system the test is being run on. Can we not wait for a selector? 

If I'm getting the purpose of this function correctly from it's name (`enableToggleByAriaLabel`), the intention is to actually enable tab stops visualization, is that right? 

On thing about enabling tab stops: when we do enable tab stops (or any other visualization I think) we create an specific `div` (more precisely `<div id="accessibility-insights-root-container">`). If the target page is created fresh for this test every time, then there is no such `div` until we enable tab stops visualization. Is this not possible?